paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -382,6 +394,35 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                     label = args.googlePayConfig?.label,
                 )
             }
+        } else if (
+            paymentSelection is PaymentSelection.New.GenericPaymentMethod &&
+            paymentSelection.paymentMethodCreateParams.typeCode == PaymentMethod.Type.BacsDebit.code
+        ) {
+            val overrideParams = paymentSelection.paymentMethodCreateParams
+
+            val bacsDebit = PaymentMethodCreateParams.createBacsFromParams(overrideParams)
+            val name = PaymentMethodCreateParams
+                .getNameFromParams(overrideParams)
+            val email = PaymentMethodCreateParams
+                .getEmailFromParams(overrideParams)
+
+            if (bacsDebit != null && name != null && email != null) {
+                bacsMandateConfirmationLauncher?.launch(
+                    BacsMandateConfirmationContract.Args(
+                        nameOnAccount = name,
+                        email = email,
+                        accountNumber = bacsDebit.accountNumber,
+                        sortCode = bacsDebit.sortCode,
+                        appearance = config?.appearance
+                    )
+                )
+            } else {
+                resetViewState(
+                    userErrorMessage = getApplication<Application>()
+                        .resources
+                        .getString(R.string.stripe_something_went_wrong)
+                )
```

Comment:
`PaymentSheet` should never result in an error state for Bacs since we disable the `PayButton` until all the information is entered. Should we still add a test for this anyways along with a custom error message for Bacs?